src/analytic_engine/src/engine.rs

```diff
@@ -234,6 +242,63 @@ impl TableEngine for TableEngineImpl {
 
         self.close_tables_of_shard(close_requests).await
     }
+
+    async fn report_statistics(&self) -> Result<Option<TableEngineStats>> {
```

Comment:
I think of one issue when using prometheus metrics to collect stats here, when a shard is transferred to another node, this metrics will remain, so you may collect those table twice.

Reply:
This case has been taken into consideration, and the solution is that a filter based on shards will be applied when reporting to meta.

Reply:
OK, a better way I can think of is to register those metrics when open shard, and unregister them when close shard.